liteflow-labs/starter-kit
# wagmi wallet integration

### Description

Integrate wagmi as wallet provider instead of the previous web3 react.
Added better management of user's connection to only be connected on the platform only when the wallet is connected AND the wallet is logged in to the API thanks to the `useAccount` hook.

### How to test

Connect the wallet on the platform on the login page or other and check that every feature work as expected.
Test when you cleanup the cookies (you should be asked to sign again)
Test to disconnect your wallet
Test to logout on the platform
Test to be on the wrong network
...

### Known issues

~~When refreshing from some pages (like the explore page) while being connected, the `onConnect` from wagmi is not triggered making the login flow "stuck"
Still debugging this part~~ Done thanks to @ismailToyran
